API.adoc

```diff
@@ -0,0 +1,402 @@
+in a single operation. Each CRUD action requires that the User executing the
+call has at least one Role with the appropriate Authority or Authorities. The
+Authorities relevant to the Things Edge API are listed below.
+
+== Authorities
+
+* https://authorities.smartcosmos.net/metadata/create
+* https://authorities.smartcosmos.net/metadata/update
+* https://authorities.smartcosmos.net/metadata/read
+* https://authorities.smartcosmos.net/metadata/delete
+* https://authorities.smartcosmos.net/things/create
+* https://authorities.smartcosmos.net/things/update
+* https://authorities.smartcosmos.net/things/read
+* https://authorities.smartcosmos.net/things/delete
+
+*Available methods*
+
+* <<create1, Create - `POST /{type}`>>
+* <<update1, Update - `PUT /{type}/{urn}`>>
+* <<read1, Read - `GET /{type}?fields={keyNames}&page={pageNumber}&size={pageSize}`>>
+* <<read2, Read - `GET /{type}/{urn}?fields={keyNames}`>>
+* <<read3, Read - `POST /{type}/findByUrns?fields={keyNames}`>>
+* <<read4, Read - `POST /{type}/findByKeyValuePairs?page={pageNumber}&size={pageSize}`>>
+* <<delete1, Delete - `DELETE /{type}/{urn}`>>
+
+=== JSON Fields
+[width="100%",options="header",cols="1,^1,^1,^1,5"]
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String | generated, value and format depend on the persistence layer implementation | no default | will be generated if not provided
+| `urn` | String representation of a thing identifier in a common scheme, e.g. `urn:thing:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | no | will be generated if not provided
+| `type` | String | no | yes | the type of the Thing (e.g., "Employee", "Building")
+| `active` | boolean |  true | no |
+| `tenantUrn` | String |  generated | generated | The representation of the tenant identifier in a common scheme, e.g. `urn:tenant:uuid:<UUID>`. Only present in returned JSON Objects
+| [JSON metadata] | JSON | no default | false | Metadata for the thing, e.g. `name`, `description`, ..
+|====================
+
+.**URN Scheme**
+NOTE: The illustrated scheme for URNs is only for documentation purposes. There
+must not be any assumptions or expectations on the scheme in the REST layer. All
+URNs or identifiers are just `String` values in the scope of REST modules. The
+scheme must of course be compatible with the underlying data store.
+
+=== Request Parameters
+[width="100%",options="header"]
+|====================
+| Parameter | Parameter Type | Format | Default | Description
+| `type` | url | String | no default | the type of the Thing (e.g., "Employee", "Building")
+| `urn` | url | String | no default | the URN of the Thing
+| `fields` | query | Comma-separated String list  |  query parameter, not set by default | Metadata to include in the response (defaults to all if not set)
+| `page` | query | String representation of Long  | 1 | page number of multi-element paged returns, starting at 1
```

Comment:
What do you think about the following? Note the `^` forces centering for just the values in individual fields.
````
| `page` | query | String representation of Long  ^| 1 | page number of multi-element paged returns, starting at 1
| `size` | query | String representation of Long  ^| 20 | number of elements per page in multi-element paged responses
| `sortOrder` | query | String of {"asc", "desc"}  ^| "asc" | sort order for paged responses
````